packages/app/src/runs/useProjectRuns.ts

```diff
@@ -0,0 +1,128 @@
+import { gql, useMutation, useQuery } from '@urql/vue'
+import { Ref, computed, onMounted, onUnmounted } from 'vue'
+import { RunsDocument, RunsContainer_FetchNewerRunsDocument, RunCardFragment } from '../generated/graphql'
+import type { RunsComposable } from './RunsComposable'
+
+gql`
+query Runs {
+  ...RunsContainer
+}`
+
+gql`
+fragment RunsContainer_RunsConnection on CloudRunConnection {
+  nodes {
+    id
+    ...RunCard
```

Comment:
Not a blocker but relevant to the continued development of GraphQL patterns - we've got two composables tied to the `RunCard` fragment, which is defined in UI component. My first instinct was it's not good to couple the data fetching layer to the component, but I'm starting to think this is actually the right pattern - the way we use GraphQL, the fragment *is* tied to the UI component, so it might actually make sense here. 

It's definitely a bit counter intuitive at first, you really need to rethinking a lot of what you'd do with a traditional REST API when working with GraphQL.

Reply:
I am taking a liking to the pattern when components are self contained and define/manage their own data, especially with the subscriptions and useSubscription composable.